cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -987,13 +1025,39 @@ private V1PodSpec createPodSpec(Type resourceType, Location runtimeConfigLocatio
    */
   private V1PodSpec createPodSpec(Type resourceType, Location runtimeConfigLocation,
                                   Map<String, RuntimeSpecification> runtimeSpecs,
-                                  String restartPolicy, List<String> args, V1VolumeMount... extraMounts) {
+                                  String restartPolicy, List<String> args, V1VolumeMount... extraMounts)
+    throws ApiException {
     V1Volume podInfoVolume = createPodInfoVolume(podInfo);
     V1Volume workDirVolume = new V1Volume().name("workdir").emptyDir(new V1EmptyDirVolumeSource());
+    List<V1Volume> additionalVolumes = new ArrayList<>();
+    List<V1VolumeMount> additionalVolumeMounts = new ArrayList<>();
 
-    String workDir = "/workDir-" + twillRunId.getId();
+    if (resourceType.equals(V1Job.class)) {
+      // Create per-run configmaps for cConf and hConf by copying over the configmaps from the default namespace
```

Comment:
should not be copying config maps. This is assuming special knowledge about how the Kubernetes environment is setup and special knowledge about the config map names.

Isn't everything that needs to be localized specified already in the collection of LocalFiles from the twill spec?